src/pages/resultsView/querySummary/QuerySummary.tsx

```diff
@@ -18,26 +18,21 @@ import {ShareUI} from "./ShareUI";
 import {ServerConfigHelpers} from "../../../config/config";
 import AppConfig from "appConfig";
 import {StudyLink} from "../../../shared/components/StudyLink/StudyLink";
+import {createQueryStore} from "../../home/HomePage";
 
 @observer
-export default class QuerySummary extends React.Component<{ queryStore:QueryStore, routingStore:ExtendedRouterStore, store: ResultsViewPageStore }, {}> {
+export default class QuerySummary extends React.Component<{ routingStore:ExtendedRouterStore, store: ResultsViewPageStore }, {}> {
 
-    @observable private queryFormVisible = false;
-    @observable private queryStoreInitialized = false;
+    @observable.ref queryStore: QueryStore | undefined;
 
     constructor() {
         super();
-        this.handleModifyQueryClick = this.handleModifyQueryClick.bind(this);
     }
 
+    @autobind
     private handleModifyQueryClick() {
-
         // this will have no functional impact after initial invocation of this method
-        this.queryStoreInitialized = true;
-
-        // toggle visibility
-        this.queryFormVisible = !this.queryFormVisible;
-
+        this.queryStore = (this.queryStore) ? undefined : createQueryStore();
```

Comment:
so if I click on modify query over and over again, this.queryStore will toggle between undefined and defined? or do you mean `this.queryStore = this.queryStore || createQueryStore()`

Reply:
oh I get it, you're using the existence of a query store to determine whether the form is visible. honestly that seems a little weird to me. how about at least we do
```
@computed get queryFormVisible() {
    return !!this.queryStore;
}
```

Reply:
sure